### PR TITLE
Fixes #766. OWTF will never run within its own directory.

### LINKED ISSUE #766: OWTF installer recommendation misleads new users
OWTF should never, ever, be run from its own directory, this makes absolutely no sense since that will result in creating the output files for a given pentest within the OWTF directory.

In the real world, an OWTF user should run OWTF *always* like:

cd /path/to/pentest/directory
/path/to/owtf.py

This results in the output files being saved in the correct directory for a given pentest, instead of the OWTF directory.

Unfortunately, the installer is recommending users to do exactly this, which has led to confusion in the past, this should be replaced with an explanation like above:

[*] Start OWTF by running './owtf.py' in parent directory

We had even distro maintainers trying to call OWTF like this, maybe we should even go a step further and literally refuse to run OWTF when it is on its own directory.

Thoughts? :)

### PR DESCRIPTION
**Pull request by affinity7**
The OWTF will exit if executed from its directory

## Description

Added a check confirming the OWTF is not executed from its directory.

## Related Issue

#766  #746 

## Reviewers

@delta24 @DarKnight24 @7a 

## How Has This Been Tested?

Tested it by running it from different directories.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Other

### Checklist:

- [x] My code follows the code style (modified PEP8) of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: owtf.py [hunk 1]
```diff
@@ -255,4 +255,9 @@ def main(args):
 
 
 if __name__ == "__main__":
+
```
Comment (DarKnight24) [inline owtf.py]: Do not add line after the function name. Remove this extra line.

File: owtf.py [hunk 2]
```diff
@@ -255,4 +255,9 @@ def main(args):
...
+    if os.path.dirname(os.path.abspath(sys.argv[0])) == os.getcwd():
```
Comment (DarKnight24) [inline owtf.py]: Instead of this check, check if OWTF is ran from its directory/sub-directories. Check for the substring, @delta24 what is your view?

Comment (delta24) [inline owtf.py]: @DarKnight24 what substring?

Comment (DarKnight24) [inline owtf.py]: If executed file folder is substring of current working directory, it is so then we can say that OWTF is ran from its directory or subdirectory.

Comment (delta24) [inline owtf.py]: ok, that makes sense.

File: owtf.py [hunk 3]
```diff
@@ -255,9 +255,7 @@ def main(args):
 
 
 if __name__ == "__main__":
-
-    if os.path.dirname(os.path.abspath(sys.argv[0])) == os.getcwd():
+    if os.getcwd().find(os.path.dirname(os.path.abspath(sys.argv[0])))==0:
```
Comment (delta24) [inline owtf.py]: Spaces around the comparison missing.

File: owtf.py [hunk 4]
```diff
@@ -255,4 +255,7 @@ def main(args):
 
 
 if __name__ == "__main__":
+    if os.getcwd().find(os.path.dirname(os.path.abspath(sys.argv[0])))==0:
+        usage("Start owtf in a different directory. 'cd path/to/pentest/directory; ./path/to/owtf.py'")
+        exit(-1)
```
Comment (delta24) [inline owtf.py]: Use `sys.exit` here. (exit is a helper for the interactive shell - sys.exit is intended for use in programs.)

**Comment by delta24** [comment]
@affinity7 Thank you for the PR :)!

**Pull request merged**